### DISCUSSION TITLE
Can multiple constraints be combined?

### DISCUSSION THREAD

**Discussion by zjiayao** [fairlearn/fairlearn#1049] [Q&A] [answered]
Hi guys,

Thanks for the fantastics package! It is really intuitive to use and the example notebooks are very helpful to get started.

I have a question -- is it possible to combine multiple constraints (e.g., DemographicParity and EqualizedOdds) together when using say Exponentiated Gradient method? I did some search but was not able to find any documentation on this; is it advisable to subclass ``fairlearn.reductions.Moment`` for this purpose?

Thank you!

**Comment by MiroDudik** [comment, marked as answer]
Thanks for kind words! We currently do not support what you're describing, but technically this is definitely possible.

You could try to create your own moment by combining the code from `EqualizedOdds` and `DemographicParity`, but that might not be the easiest route (you'd probably need to extend the functionality of `UtilityRateParity` to work with events that are not mutually exclusive).

I think technically cleanest would be to subclass `fairlearn.reductions.ClassificationMoment` into a new class called something like `Intersect` (or perhaps a better name), which would be constructed by invoking `Intersect(moment1, moment2)`, where `moment1` and `moment2` would be arbitrarily initialized instances of `fairlearn.reductions.ClassificationMoment`. You could then write things like:
```
expgrad = ExponentiatedGradient(
    LogisticRegression(),
    Intersect(EqualizedOdds(difference_bound=0.05), DemographicParity(difference_bound=0.10)
)
```

**Reply by zjiayao** [reply]
Thank you, Miro! I'll give that a try :)
